CONTRIBUTING.md

```diff
@@ -1,39 +1,145 @@
+  for more details.
+
+# Contributing Code
+
+## Git workflow
+
+The preferred workflow for contributing to Graspologic is to fork the main repository on GitHub, clone, and develop on a
+branch. Steps: 
+
+1. Fork the [project repository](https://github.com/microsoft/graspologic) by clicking on the ‘Fork’ button near the top
+   right of the page. This creates a copy of the code under your GitHub user account. For more details on how to
+   fork a repository see [this guide](https://help.github.com/articles/fork-a-repo/).
+
+2. Clone your fork of the Graspologic repo from your GitHub account to your local disk:
+
+   ```bash
+   git clone git@github.com:YourGithubAccount/graspologic.git
+   cd graspologic
+   ```
+
+3. Create a feature branch to hold your development changes:
+
+   ```bash
+   git checkout -b my-feature
+   ```
+
+   Always use a `feature` branch. Pull requests directly to either `dev` or `main` will be rejected
+   until you create a feature branch based on `dev`.
+
+4. Develop the feature on your feature branch. Add changed files using `git add` and then `git commit` files:
+
+   ```bash
+   git add modified_files
+   git commit
+   ```
+
+   After making all local changes, you will want to push your changes to your fork:
+   ```bash
+   git push -u origin my-feature
+   ```
+
+## Pull Request Checklist
+
+We recommended that your contribution complies with the following rules before you submit a pull request: 
+
+- Follow the [coding-guidelines](guidelines). 
+- Give your pull request a helpful title that summarizes what your contribution does. 
+- Link your pull request to the issue (see: 
+  [closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) 
+  for an easy way of linking your issue)
+- All public methods should have informative docstrings with sample usage presented as doctests when appropriate. 
+- At least one paragraph of narrative documentation with links to references in the literature (with PDF links when 
+  possible) and the example. 
+- If your feature is complex enough that a doctest is insufficient to fully showcase the utility, consider creating a 
+  Jupyter notebook to illustrate use instead
+- All functions and classes must have unit tests. These should include, at the very least, type checking and ensuring 
+  correct computation/outputs.
+- All functions and classes should be rigorously typed with Python 3.5+ 
+  [`typehinting`](https://docs.python.org/3/library/typing.html). Validate your typehinting by running `mypy ./graspy`
+- All code should be automatically formatted by `black`. You can run this formatter by calling:
+  ```bash
+  pip install black
+  black path/to/your_module.py
+  ```
+- Ensure all tests are passing locally using `pytest`. Install the necessary
+  packages by: 
+
+  ```bash
+  pip install pytest pytest-cov
+  pytest
+  ```
+
+# Guidelines
+----------
```

Comment:
```suggestion
```
I see this is copied from the `docs` and not sure why I have this, but its not needed.